### PR TITLE
Added a feature which disables the clear button when nothing is selected.

### LINKED ISSUE #5500: Clear/Reset Filters button on the map page should be disabled by default
### Description

Right now, Clear/Reset filters is enabled by default, and clicking on it doesn't do anything. When I click on a tag, and then click on this button, it works fine.
I think disabling it by default would be a better UX. Else, the user is confused when nothing happens when the button is clicked without any tags being selected.

### Screenshots

Screenshot 2023-03-17 at 6 26 34 PM

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vedant-Manjrekar**
## Fixes Issue #5500

closes #5500

## Changes proposed

Currently `Clear/Reset` button in Map section is active even when no filter is selected, hence a fix to this issue is proposed through my PR which disables the button if no filters are selected.

Before:

Screenshot 2023-03-21 at 8 43 00 PM

After:

Screenshot 2023-03-21 at 8 45 31 PM

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5588"></a>

### CODE REVIEW

File: pages/map.js [hunk 1]
```diff
@@ -56,6 +56,8 @@ export default function Map({ data }) {
   const [filteredUsers, setFilteredUsers] = useState([]);
   const [selectedTags, setSelectedTags] = useState(new Set());
 
+  console.log(selectedTags);
```
Comment (PiseAtharva) [inline pages/map.js:59]: Console Logs are alright while debugging but best remove them after your done (standard production level project guidlines)

Comment (Vedant-Manjrekar) [reply pages/map.js]: Sorry, my bad. It's fixed now.

File: components/Button.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default function Button({
     ? " text-white bg-indigo-600  hover:bg-indigo-700"
     : " text-indigo-600 hover:bg-indigo-50";
   className += disable
-    ? " border-2 border-red-600 hover:bg-red-600"
+    ? " border-2 border-red bg-[#f4f4f4] hover:bg-[#f3f3f3] opacity-[.5] text-[gray] cursor-not-allowed"
```
Comment (eddiejaoude) [inline components/Button.js:15]: I think we can use the standard tailwind colours?

**Review by eddiejaoude — APPROVED** [review]
Looks great, thanks 👍

File: components/Badge.js [hunk 3]
```diff
@@ -36,7 +37,11 @@ export default function Badge({
   const badge = (
     <div
       title={title}
-      className={`absolute inline-block rotate-0 skew-x-0 skew-y-0 scale-x-100 scale-y-100 py-1 px-1.5 text-xs leading-none text-center whitespace-nowrap align-baseline font-bold bg-orange-600 text-black rounded-full z-10 ${css} ${badgeClassName}`}
+      className={`absolute inline-block rotate-0 skew-x-0 skew-y-0 scale-x-100 scale-y-100 py-1 px-1.5 text-xs leading-none text-center whitespace-nowrap align-baseline font-bold bg-orange-600 ${
+        disable ? "text-[#EEEEEE]" : "text-black"
+      } ${
+        disable ? "bg-[#C2C2C2]" : "bg-orange-600"
```
Comment (eddiejaoude) [inline components/Badge.js:43]: I think both `disable`s can be combined into one? I will fix via temporary branch

**Comment by Vedant-Manjrekar** [comment]
@eddiejaoude the changes you made in the PR has removed the disabled effect, now both when the button is disabled and when not looks similar.

Check it out:
Screenshot 2023-03-28 at 12 35 47 PM

It was supposed to look like this:
Screenshot 2023-03-21 at 8 43 12 PM

**Comment by eddiejaoude** [comment]
Yes, locally it worked for me, see the screenshot, but I didn't get a chance to see why when deployed it didn't work correctly

Screenshot 2023-03-28 at 14 14 53

If you have time, feel free to take a look and submit a PR fix, otherwise I can take a look tomorrow

**Comment by Vedant-Manjrekar** [comment]
> Yes, locally it worked for me, see the screenshot, but I didn't get a chance to see why when deployed it didn't work correctly
> 
> Screenshot 2023-03-28 at 14 14 53
> 
> If you have time, feel free to take a look and submit a PR fix, otherwise I can take a look tomorrow

I'll take care of it.

**Comment by eddiejaoude** [comment]
Great, thank you 👍

**Pull request merged**